### ISSUE TITLE
Distributed Tracing GA Plan

### ISSUE THREAD

**Issue by bachuv** [Azure/azure-functions-durable-extension#2071]
#### DurableTask
- [ ] Merge OpenTelemetry PR (includes removing the old implementation and adding integration tests)
- [ ] Add exception information
- [ ] Add timer information
- [ ] Add docs and samples about how to use it with App Insights, Zipkin, and other telemetry exporters

#### Durable Functions
- [ ] Consume information from DurableTask
- [ ] Add entity information
- [ ] Remove old implementation
- [ ] Add integration tests
- [ ] Add docs and samples about how to use it with App Insights, Zipkin, and other telemetry exporters

#### Out of Proc
- [ ] Test that it works “out of the box” (currently traces are correlated without any OOProc specific changes)
- [ ] Add docs and samples about how to use it with App Insights, Zipkin, and other telemetry exporters

#### Miscellaneous
- [ ] Investigate getting ILogger traces to also correlate (e.g. log.LogInformation())

**Comment by marnilss** [comment]
Is there any update on this? 

I happened to see this page: https://github.com/Azure/azure-functions-durable-extension/blob/dev/samples/correlation-csharp/getting-started.md
Is it related, and how? Is it only for a preview or is it in a released version?

**Comment by cgillum** [comment]
@marnilss the above link refers to a preview, but that preview is available in the current releases. That said, we're redoing the implementation so that it works with all our backend storage providers and so that it's compatible with OpenTelemetry. This issue tracks work required for the rewrite.

**Comment by uchitha** [comment]
@cgillum We're currently reviewing our logging to improve on distributed tracing capabilities. We got multiple durable functions taking part in business transactions that we'd like to track. All of our functions use App Insights. Do you recommend going ahead with the existing implementation at this point or waiting until the new implementation comes out? We are happy to wait if it provides us with more support/predictability. Also aware that the existing implementation lacks support for certain scenarios (like entity funcs)

**Comment by cgillum** [comment]
@uchitha I'd recommend waiting for the new implementation if you can. It will definitely provide you with better support and predictability since we want to deprecate/remove the existing previous functionality.

**Comment by marnilss** [comment]
Apparently it is in preview now !!! https://github.com/Azure/azure-functions-durable-extension/tree/dev/samples/distributed-tracing/v2/DistributedTracingSample

OR is it something else, that looks like what we have been waiting for?

**Comment by cgillum** [comment]
@marnilss yes, it's in preview now (finally)! You can find the official announcement here: https://azure.microsoft.com/en-us/updates/public-preview-distributed-tracing-v2-for-durable-functions/.

**Comment by gha-zund** [comment]
we would really love to try that out, but there are two big must-haves missing for us:
- durable entity for isolated function
- send tracing out via OpenTelemetry

we are standing in front a decision:
implement tracing all by our own (which would mean a significant time investment).
or wait until this is available (but ... for how long?)

is there a possibility to push this topic? (user voice or something?)

**Comment by lsl-2020** [comment]
> we would really love to try that out, but there are two big must-haves missing for us:
> 
> * durable entity for isolated function
> * send tracing out via OpenTelemetry
> 
> we are standing in front a decision: implement tracing all by our own (which would mean a significant time investment). or wait until this is available (but ... for how long?)
> 
> is there a possibility to push this topic? (user voice or something?)

@gha-zund For #2 item, they are working on supporting OTel now. You can redirect to the corresponding links starting from here: https://github.com/Azure/azure-functions-host/issues/9273.

**Comment by lsl-2020** [comment]
Hi @cgillum and other experts, since OpenTelemetry support PR is merged in both worker side and host side. Any updates on current thread? Do we have an ETA of when we can adopt DT in Azure Functions other than Application Insights?
- [Release v4.1033.8 · Azure/azure-functions-host (github.com)](https://github.com/Azure/azure-functions-host/releases/tag/v4.1033.8)
- [Release Microsoft.Azure.Functions.Worker 1.22.0 · Azure/azure-functions-dotnet-worker (github.com)](https://github.com/Azure/azure-functions-dotnet-worker/releases/tag/1.22.0)

**Comment by julealgon** [comment]
I'm also looking for updates on this. I'm in the process of instrumenting the last applications on our side with OTEL and stumbled upon the problem of configuring it on Azure Functions projects.

I'm a bit confused as to why this was closed:
- https://github.com/Azure/azure-functions-host/issues/9273

Is this issue here supposed to be the central place for tracking progress on OpenTelemetry support in Azure Functions?
